nixos/modules/programs/waybar.nix

```diff
@@ -5,16 +5,29 @@ with lib;
 {
   options.programs.waybar = {
     enable = mkEnableOption (lib.mdDoc "waybar");
-  };
-
-  config = mkIf config.programs.waybar.enable {
-    systemd.user.services.waybar = {
-      description = "Waybar as systemd service";
-      wantedBy = [ "graphical-session.target" ];
-      partOf = [ "graphical-session.target" ];
-      script = "${pkgs.waybar}/bin/waybar";
+    package = mkOption {
```

Comment:
Instead of using `mkOption` directly you should use `mkPackageOptionMD` instead.
 ```
 package = mkPackageOptionMD pkgs "waybar" { };
 ```